### PR TITLE
fix(tests): async tests using done function

### PR DESCRIPTION
**Pull request by bartoszmajsak**
While working on a fix for pipeline selection (coming) I noticed that bunch of async tests, where observables are involved, are never reaching `.subscribe` closure. That implies they are always green and not really bringing any value as any verification in the subscriber is never called. 

Most of the fixes were just adding `done()` call. In some places the code is wrong though. I fixed as much as I could, but `xit` three of them with `// FIXME` comment.

### CODE REVIEW

File: src/app/shared/context.service.spec.ts [hunk 1]
```diff
@@ -169,10 +169,12 @@ describe('Context Service:', () => {
     contextService.changeContext(navigation).subscribe(val => {});
     contextService.current.subscribe(val => {
       expect((val.user as any).features).toEqual(features);
+      done();
     });
   });
 
-  it('Feature-flag - getFeatures return an error', () => {
+  // FIXME
+  xit('Feature-flag - getFeatures return an error', (done: DoneFn) => {
```
Comment (bartoszmajsak) [inline src/app/shared/context.service.spec.ts]: @corinnekrych I hope you could help me fix this one.

File: src/app/space/app-launcher/services/app-launcher-gitprovider.service.spec.ts [hunk 2]
```diff
@@ -47,59 +47,67 @@ describe('Service: AppLauncherGitproviderService', () => {
...
-  it('Get GitHubDetails', () => {
-    mockService.connections.subscribe((connection: any) => {
-        connection.mockRespond(new Response(
-          new ResponseOptions({
-            body: JSON.stringify(gitHubDetails),
-            status: 200
-          })
-        ));
+  // FIXME does two http calls so we can't return the same data for both cases
+  xit('Get GitHubDetails', (done: DoneFn) => {
```
Comment (bartoszmajsak) [inline src/app/space/app-launcher/services/app-launcher-gitprovider.service.spec.ts]: @invincibleJai could you please take a look at how to fix this?

File: src/app/space/app-launcher/services/app-launcher-project-summary.service.spec.ts [hunk 3]
```diff
@@ -1,68 +1,72 @@
...
-              body: JSON.stringify({'uuid': 'e6daff35-5d93-4c38-965a-6a975cf80be1', 'uuid_link': '/status/e6daff35-5d93-4c38-965a-6a975cf80be1'}),
-              status: 200
-            })
-          ));
-      });
-      service.setup(summaryData, 3).subscribe((val: any) => {
-          expect(val).toBeDefined();
-          expect(val.uuid).toEqual('e6daff35-5d93-4c38-965a-6a975cf80be1');
+  // FIXME presumably can't mock all the http interactions with one response
+  xit('Should return uuid', (done: DoneFn) => {
```
Comment (bartoszmajsak) [inline src/app/space/app-launcher/services/app-launcher-project-summary.service.spec.ts]: @invincibleJai could you please take a look at how to fix this?

**Comment by joshuawilson** [comment]
please rebase

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.ts [hunk 4]
```diff
@@ -43,6 +43,7 @@ export class AppLauncherPipelineService implements PipelineService {
       return this.http.get(runtimeEndPoint, option)
         .map(response => response.json() as Pipeline[])
         .map(pipelines => {
+          console.log(pipelines);
```
Comment (jiekang) [inline src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: Is this needed? Prefer to be removed if possible.

Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: not at all. shame on me. we have to get some bot to catch that.

Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: I will fix that after the rebase. :beers: @jiekang

Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: Done in b5f581303d48265cc04704f46a5ce5aa88486c03

**Comment by bartoszmajsak** [comment]
@joshuawilson I merged master and resolved the conflicts. We can squash&merge from here.

**Comment by joshuawilson** [comment]
@bartoszmajsak you have outstanding questions? Do you not want them answered?

**Comment by bartoszmajsak** [comment]
How about we create a "tech-debt" issue out of those `xit`s and assign them accordingly?

**Comment by joshuawilson** [comment]
that works for me

File: src/app/shared/context.service.spec.ts [hunk 5]
```diff
@@ -182,9 +184,10 @@ describe('Context Service:', () => {
     });
 
     // when
-    contextService.changeContext(navigation).subscribe(val => {});
+    contextService.changeContext(navigation).subscribe(() => {});
```
Comment (joshuawilson) [inline src/app/shared/context.service.spec.ts]: what is the point of this? you pass nothing in , return nothing, and do nothing, right?

Comment (bartoszmajsak) [reply src/app/shared/context.service.spec.ts]: Well... I'm asking myself the same question. @corinnekrych might know?

**Pull request merged**